examples/misc/test/effective_dart_test.dart

```diff
@@ -40,8 +40,8 @@ void main() {
         // Creates a List<int>:
         var iterable = [1, 2, 3];
 
-        // Prints "List<dynamic>":
-        print(List.from(iterable).runtimeType);
+        // Prints false because the type is a List<dynamic> instead
+        print(List.from(iterable) is List<int>);
```

Comment:
I think we should leave the Effective Dart examples as they were before. It is generally a bad idea to use runtimeType in normal code, but in the example here we're using it specifically because we really do want to show the exact runtime type. Using is might confuse readers because then they have to think about the subtyping rules around generic types, which can be pretty confusing for many users.

Reply:
@munificent  - any thoughts on how to deliver the Effective Dart message without using `.runtimeType`?

I first suggested `is` because I was trying to imagine how dropping the type parameter will fail later in the program. My first thought was that it would fail when you try to assign it to a variable/parameter of the original type. So at that point you are already having to address subtyping rules to understand a cast failure.

Another point of failure is missing errors when you expected them (e.g. `List.from(iterable).add("invalid-value-type")`), but that may be more confusing.

Reply:
Oh, dear. I'm sorry, I gave completely inconsistent feedback between here and on the bug. :(

I'll follow up here and on the bug.